Added tests and updated requirements

andrewmoorman opened the pull request:
requirements.txt now includes update to scipy and leidenalg

new tests folder added with 7 unit tests for functions which rely on scipy/leidenalg

cluster function in cluster.py refactored to separate community detection functions for testing

Note: Maybe we should create a "version 1.5.7" branch as mentioned?

andrewmoorman committed:
Added tests and updated requirements

requirements.txt now includes update to scipy and leidenalg

new tests folder added with 7 unit tests for functions which rely on scipy/leidenalg

cluster function in cluster.py refactored to separate community detection functions for testing

andrewmoorman committed:
updated requirements

leidenalg updated to 0.8.2

andrewmoorman committed:
Phenograph tutorial

Tutorial on how to use Phenograph with attention paid to choosing parameter values and (broadly) explaining what's happening "behind the scenes."

Accompanying 10x dataset and human RB genes list is also included

andrewmoorman committed:
Tutorial update

Removed cell typing material

hisplan commented:
I see two `RB_genes_human.txt`. Let's remove one of them. Maybe keep the one in the `data` folder?

andrewmoorman committed:
updated tutorial and fixed pytest warnings

Tutorial now instructs users to import scanpy, which isn't a requirement of Phenograph

Classify module doesn't create deprecation warning for scipy.sparse.linalg

hisplan commented:
Hmm. I'm not sure if your 1.5.7 branch is fully synced with dpeerlab:1.5.7.

andrewmoorman committed:
Delete RB_genes_human.txt

andrewmoorman committed:
Merge remote-tracking branch 'upstream/1.5.7' into 1.5.7

andrewmoorman commented:
Should be okay now

hisplan committed:
Remove .DS_Store

hisplan committed:
Update .gitignore

hisplan committed:
Update .gitignore

hisplan committed:
Reformat

hisplan commented:
@armMSKCC a few comments:

Code
- Bump up the version to 1.5.7 in `version.py`
- Run Black on `test_cluster.py` (https://github.com/psf/black). You can also run this from VS Code or PyCharm.

Notebook
- Change Phenograph to PhenoGraph.
- Add `phenograph.__version__` to display PhenoGraph version along with Scanpy versions.
- `pip install scanpy==1.6.0` - the notebook might not run with the old versions.
- Rename `Phenograph_tutorial.ipynb` to `tutorial-pbmc3k.ipynb`
- Rename `example-01-using-random-data-points.ipynb` to `example-using-random-data-points.ipynb`

README.md
- Update release notes.
- Add a new section and describe how one can run pytest and resolve the issue we encountered.
- Mention and add a link to the new tutorial.

hisplan commented:
@armMSKCC Forgot one thing. The notebook failed to load the RB file.

andrewmoorman committed:
File renaming and updates to README

Changed files names as appropriate, ran black on test_classify.py, updated version, updated README

andrewmoorman committed:
Merge branch '1.5.7' of https://github.com/armMSKCC/PhenoGraph into 1.5.7

andrewmoorman committed:
Update test_cluster.py

andrewmoorman committed:
Updated README

hisplan commented on README.md:
```diff
@@ -49,6 +49,11 @@ If you use PhenoGraph in work you publish, please cite our publication:
 
+### Version 1.5.7
+
+* Updated leidenalg and scipy version requirements, revised parallel jaccard to support scipy==1.5.1, and created a test collection for use with pytest (see above)
```
"see above" --> "see below" you mean? Maybe make a link to the unit test section?

hisplan commented on README.md:
```diff
@@ -127,3 +151,11 @@ Troubleshooting
+
+* When running unit tests, pytest may deliver the following warning `RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility`. This is caused by an incompatability between newer versions of the pytest module and an older version of Python. Check your Python version with:
+
+    python --version
```
Put this into a preformatted box like this:
```
python --version
```

hisplan reviewed (changes requested):
Minor comments

hisplan commented:
I still don't see these in the Jupyter notebook:
- Add `phenograph.__version__` to display PhenoGraph version along with Scanpy versions.
- `!pip install scanpy==1.6.0` - the notebook might not run with the old versions.

andrewmoorman committed:
Updated tutorial file and README

Tutorial works with new RB file location and includes requested edits, README links to pytest section

andrewmoorman committed:
Removed jupyter notebook scanpy output

Removed my scanpy installation output from jupyter notebook

andrewmoorman committed:
Removed outdated tutorial and edited newly named version

andrewmoorman committed:
Update README.md

andrewmoorman committed:
Update README.md

andrewmoorman committed:
Update tutorial_pbmc3k.ipynb
